cmd/skopeo/inspect.go

```diff
@@ -86,21 +90,37 @@ func (opts *inspectOptions) run(args []string, stdout io.Writer) (retErr error)
 		return err
 	}
 
-	img, err := parseImage(ctx, opts.image, imageName)
+	sys, err := opts.image.newSystemContext()
+	if err != nil {
+		return err
+	}
+
+	src, err := parseImageSource(ctx, opts.image, imageName)
 	if err != nil {
 		return err
 	}
 
+	closer := io.Closer(src)
+
 	defer func() {
-		if err := img.Close(); err != nil {
+		if err := closer.Close(); err != nil {
 			retErr = errors.Wrapf(retErr, fmt.Sprintf("(could not close image: %v) ", err))
 		}
 	}()
 
-	rawManifest, _, err := img.Manifest(ctx)
+	rawManifest, _, err := src.GetManifest(ctx, nil)
 	if err != nil {
 		return err
 	}
+
+	if opts.config || !opts.raw {
+		img, err = image.FromSource(ctx, sys, src)
+		if err != nil {
+			return err
```

Comment:
(This is all top-level CLI; please wrap errors in user-relevant explanations about the context, like the code below does.)